src/mem.c

```diff
@@ -89,12 +89,19 @@ kore_realloc(void *ptr, size_t len)
 void *
 kore_calloc(size_t memb, size_t len)
 {
-	if (memb == 0 || len == 0)
+	void *result;
+	const size_t nbytes = memb * len;
+
+	if (nbytes == 0)
 		fatal("kore_calloc(): zero size");
-	if (SIZE_MAX / memb < len)
```

Comment:
Hi,

Thanks for this, I never got around writing it and wasn't particularly happy with valgrind being angry at me :-)

Why did you remove the overflow check in kore_calloc() though? It is now susceptible to an integer overflow. The check you introduced is just plain wrong.

Rest looks promising. Although I am not a big fan of the fini naming convention.

.joris